agent/structs/config_entry.go

```diff
@@ -479,7 +479,12 @@ func (c *ConfigEntryResponse) MarshalBinary() (data []byte, err error) {
 	bs := make([]byte, 128)
 	enc := codec.NewEncoderBytes(&bs, msgpackHandle)
 
-	if err := enc.Encode(c.Entry.GetKind()); err != nil {
+	kind := ""
+	if c.Entry != nil {
```

Comment:
when is it actually valid to have a nil entry here?

Reply:
A read against a non-existant entry

Reply:
I guess maybe we could return an error instead but it would be harder to translate to the proper HTTP response code as we would have to do string inspection of the error which may change between versions of consul and could not work if clients do not get updated as quickly as the servers.